Sources/SwiftParser/TokenPrecedence.swift

```diff
@@ -192,47 +192,266 @@ enum TokenPrecedence: Comparable {
     case  // Literals
     .Self, .false, .nil, .`self`, .super, .true:
       self = .identifierLike
+
     // MARK: Expr keyword
     case  // Keywords
-    .as, .is, .try,
+    .as, .is, .some, .try,
+      .await, .reasync, .async,
       // We don't know much about which contextual keyword it is, be conservative an allow considering it as unexpected.
       // Keywords in function types (we should be allowed to skip them inside parenthesis)
       .rethrows, .throws,
       // Consider 'any' a prefix operator to a type and a type is expression-like.
       .Any,
       // 'where' can only occur in the signature of declarations. Consider the signature expression-like.
       .where,
+      .each,
       // 'in' occurs in closure input/output definitions and for loops. Consider both constructs expression-like.
       .in:
       self = .exprKeyword
+
     case  // Control-flow constructs
     .defer, .do, .for, .guard, .if, .repeat, .switch, .while,
       // Secondary parts of control-flow constructs
       .case, .catch, .default, .else,
       // Return-like statements
       .break, .continue, .fallthrough, .return, .throw, .then, .yield:
       self = .stmtKeyword
+
     // MARK: Decl keywords
     case  // Types
-    .associatedtype, .class, .enum, .extension, .protocol, .struct, .typealias,
+    .associatedtype, .class, .enum, .extension, .protocol, .struct, .typealias, .actor, .Sendable,
       // Access modifiers
       .fileprivate, .internal, .private, .public, .static,
       // Functions
       .deinit, .func, .`init`, .subscript,
       // Variables
-      .let, .var,
+      .let, .var, .weak,
       // Operator stuff
       .operator, .precedencegroup,
       // Declaration Modifiers
       .__consuming, .final, .required, .optional, .lazy, .dynamic, .infix, .postfix, .prefix, .mutating, .nonmutating, .convenience, .override, .package, .open,
-      .__setter_access, .indirect, .nonisolated, .distributed, ._local,
-      .inout, ._mutating, ._borrowing, ._consuming,
+      .__setter_access, .indirect, .isolated, .nonisolated, .distributed, .unchecked, ._local,
+      .inout, ._mutating, ._borrow, ._borrowing, .borrowing, ._consuming, .consuming, .consume,
+      ._backDeploy, .backDeployed, .inline, ._specialize, .availability, .unavailable, .deprecated, ._spi,
```

Comment:
What was the reasoning to change these to `declKeyword`? For example `_spi` should be an `exprKeyword` since it’s a type attribute name and when parsing a type attribute we are usually looking for an identifier that specifies the variable name, which is not an introducer of a new declaration.

That being said, I think it might make sense to open a section for the cases in `TypeAttribute` and one for the cases in `DeclarationAttributeWithSpecialSyntax` just to group them together.

Reply:
Added a new group for `DeclarationAttributeWithSpecialSyntax`.
Not sure tough if it have the right precedence. 
Name can also be discussed 😬

Question: Why is `.rethrows` in here? Does that have special syntax?

b4c0a3d7dada89146f2f8f615db9fc96e2371c42

Reply:
Added new `typeKeyword` group in 59d1c9383174c4bb2e4873b30c80ed3d16d0efd9

Again here some keyword are already used. 
Not sure where to place them. 


Same if they have the right precedence.

Reply:
Oh, I think you misunderstood me. I didn’t mean to introduce a new precedence level, just to have a separate `case` clause in the `switch` for all the `TypeAttribute`s and cases in `DeclarationAttributeWithSpecialSyntax`.

> Question: Why is .rethrows in here? Does that have special syntax?

There’s the `@rethrows` and I think we would be rejecting that otherwise because the parser expects an identifier after `@` and `rethrows` is a laxer-classified keyword.

Reply:
Oh sorry about that. 
I've added them as decl. keyword now. 
But I'm not sure where the redundant keyword should be placed tough? 

1. `.async` and `._local` belongs both to `declKeyword`. 
    But what group to you think it's the best fit?
2. `rethrows` and `differentiable` is also in two groups.
    Again, what do you think is the best fit?

Reply:
If in doubt, I would put them in the non-attribute section. IMO the `TypeAttribute` and `DeclarationAttributeWithSpecialSyntax` sections are just to keep the `Other` group slightly smaller.

Also, I would change `TypeAttribute` and `DeclarationAttributeWithSpecialSyntax` to return `exprKeyword` instead of `declKeyword`. That’s what I meant to say by

> For example `_spi` should be an `exprKeyword` since it’s a type attribute name and when parsing a type attribute we are usually looking for an identifier that specifies the variable name, which is not an introducer of a new declaration.

Reply:
Ah okay, think it starts to make sense now. 
Thanks, and sorry for the confusion 😁

Reply:
Not a problem at all. Thanks for doing this.